### ISSUE TITLE
Incorrect boto.vpc.routetable.RouteTable objects with propagated routes

### ISSUE THREAD

**Issue by ojongerius** [boto/boto#3049]
When investigating a boto.vpc.routetable.RouteTable object I noticed the following behaviour:

If the route table has route propagation enabled, we see 'half baked' route table objects being created, containing just tags. The behaviour is:

If no route propagation is enabled; create a correct route table object.
If route propagation _is_ enabled; create a route object with tags set to an empty dict ({}). Next, for each tag in the tagSet; create a route object containing no routes, and 'key' set to the key of the fist tag, 'value', set to the value of the first tag.

Ipython example:

Normal Route object:

```
In [7]: vars(all_route_tables[1].routes[2])
Out[7]:
{'destination_cidr_block': u'1.2.3.4/16',
 'gateway_id': u'vgw-123',
 'instance_id': None,
 'interface_id': None,
 'state': u'active',
 'vpc_peering_connection_id': None}
```

Route object with route propagation, notice tags are empty:

```
In [9]: vars(all_route_tables[1])
Out[9]:
{'associations': [RouteAssociation:rtbassoc-123],
 'connection': VPCConnection:ec2.us-east-1.amazonaws.com,
 'gatewayId': u'vgw-123',
 'id': u'rtb-123',
 'item': u'\n                ',
 'region': RegionInfo:us-east-1,
 'routes': [Route:1.2.3.4/17,
  Route:0.0.0.0/0,
  Route:1.2.3.5/16,],
 'tags': {},
 'vpc_id': u'vpc-123'}
```

The next Route object contains the tags of the previous one:

```
{'associations': [],
 'connection': VPCConnection:ec2.us-east-1.amazonaws.com,
 'id': None,
 'item': u'\n                ',
 'key': u'Foo',
 'region': RegionInfo:us-east-1,
 'routes': [],
 'tags': {},
 'value': u'Bar',
 'vpc_id': None}
```

And so on.

PR #3046 fixes this issue, and adds tests for it. PR #3047 contains the fix and adds a test for the route origin.

**Comment by ojongerius** [comment]
Anything I can do so we can get this merged?

**Comment by tomkite** [comment]
How do I get this fix? I have botocore 1.1.2 and boto 2.38 but still see empty tags.

**Comment by ojongerius** [comment]
Guessing this will never happen. Closing time.
